backend/app/controllers/books_controller.rb

```diff
@@ -5,16 +5,18 @@ class BooksController < ApplicationController
   before_action :authorize_request
 
   def index
-    render(json: CollectionPresenter.new(books), status: :ok)
+    render(json: CollectionPresenter.new(books, BookSerializer), status: :ok)
   end
 
   def show
-    render(json: { data: book }, status: :ok)
+    render(json: { data: book ? BookSerializer.new(book) : nil }, status: :ok)
   end
 
   def create
+    @book = Book.create(book_params)
```

Comment:
I think you can put this in the book method and use params[:id] as a conditional. Just test it so you know what to expect.